src/main/java/org/symphonyoss/integration/jira/authorization/JiraAuthorizationManager.java

```diff
@@ -376,7 +373,7 @@ public void authorizeTemporaryToken(IntegrationSettings settings, String tempora
    * @return JiraOAuth1Provider configured.
    * @throws JiraOAuth1Exception Thrown in case of error.
    */
-  private JiraOAuth1Provider getJiraOAuth1Provider(IntegrationSettings settings, String baseUrl)
+  public JiraOAuth1Provider getJiraOAuth1Provider(IntegrationSettings settings, String baseUrl)
       throws JiraOAuth1Exception {
     AppAuthorizationModel appAuthorizationModel = getAuthorizationModel(settings);
```

Comment:
We are reading public/private key on the filesystem on every request. It's a lot of unnecessary I/O operations.